homeassistant/helpers/template.py

```diff
@@ -1470,6 +1471,23 @@ def state_attr(hass: HomeAssistant, entity_id: str, name: str) -> Any:
     return None
 
 
+def has_value(hass: HomeAssistant, entity_id: str | State) -> bool:
+    """Test if an entity has a valid value."""
+    value: str | None
+    if isinstance(entity_id, str):
+        if state_obj := _get_state(hass, entity_id):
+            value = state_obj.state
+        else:
+            value = None if valid_entity_id(entity_id) else entity_id
+
+        if value is None or value in [STATE_UNAVAILABLE, STATE_UNKNOWN]:
+            return False
+    elif entity_id.state in [STATE_UNAVAILABLE, STATE_UNKNOWN]:
+        return False
+
+    return True
```

Comment:
Considering all other methods, they don't accept state objects either. I think this method can be as simple as:

```suggestion
def has_value(hass: HomeAssistant, entity_id: str) -> bool:
    """Test if an entity has a valid value."""
    state_obj = _get_state(hass, entity_id)
    return state_object is not None and (
        state_obj.state not in [STATE_UNAVAILABLE, STATE_UNKNOWN]
    )
```

Reply:
So I removed STATE object accordingly. 

I did leave the check in there that if _get_state returns None to check if it is a valid entity ID and if not then to just check the string.

This allows for like checking if an attribute exist:
```has_value((state_attr('sensor.train', 'arrival') or "unknown"|string))```

which then allows one to do something like:
```{{ ([states('binary_sensor.charging'), (state_attr('sensor.battery_state', 'charging') or "unknown")|string)] | select('has_value') | first) in ['on','True'] }}```
to use the state of the entity if it is available, and if not to use the attribute of another entity as fall-back.

Reply:
> as_value((state_attr('sensor.train', 'arrival') or "unknown"|string))

That is not a concern for this method, as we also do not consider it a concern in all other method. Please apply the change as suggested.

Reply:
Understand it is not a concern. What is the proposal then for items like that?